datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -260,33 +260,43 @@ impl BatchPartitioner {
                 } => {
                     let idx = *next_idx;
                     *next_idx = (*next_idx + 1) % *num_partitions;
-                    Box::new(std::iter::once(Ok((idx, batch))))
+                    assert_eq!(batches.len(), 1);
```

Comment:
should we have 
```
mut batches: Vec<RecordBatch>
```

if ```assert_eq!(batches.len(), 1)```
?

Reply:
For round robin, `batches.len` is always 1 (no buffering needed)/

Reply:
Thanks @Dandandan what I mean do we really need a vector of batches as input if there is only 1 batch always ?